### PR TITLE
feat(buildkite/download-artifact): support download artifacts from Buildkite

### PR DESCRIPTION
**Pull request by v1v**
Specialised GitHub action to download a given set of artifacts from a given Buildkite pipeline build.

I decided to use individual actions rather than altogether as we used to have in https://github.com/elastic/apm-pipeline-library/tree/main/.github/actions/buildkite

### CODE REVIEW

File: buildkite/run/README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Following inputs can be used as `step.with` keys
 | `branch`     | String  | `main`                                              | Branch the commit belongs to. This allows you to take advantage of your pipeline and step-level branch filtering rules. |
 | `commit`     | String  | `HEAD`                                              | Ref, SHA or tag to be built.                                                                                      |
 | `message`    | String  | `"Triggered automatically - ${{ github.workflow_ref }} (${{ github.event_name }})"`| The Buildkite build message to be shown in the UI.  |
+| `token`      | String |                                             | The Buildkite API Token.                                                                          |
```
Comment (v1v) [inline buildkite/run/README.md:19]: unrelated

File: buildkite/download/action.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (reakaleek) [inline buildkite/download/action.yml:1]: What do you think about the name `buildkite/download-artifacts`?

Comment (reakaleek) [reply buildkite/download/action.yml:1]: I guess it could be renamed to `path`?

```yaml
- uses: elastic/oblt-actions/buildkite/download-artifacts@v1
        with:
          path: "artifacts.tar"
          build: ${{ steps.buildkite-run.outputs.build }}
          pipeline: "my-super-pipeline"
          token: ${{ secrets.BUILDKITE_TOKEN }}
```

Only "download" sounds so ambiguous. 🤷 

also if you plan to create an action called `buildkite/log` then I think `buildkite/artifacts` sounds like a better fit.

but generally.. I think `buildkite/download-artifacts` is explicit and doesn't leave ambiguity

Comment (v1v) [reply buildkite/download/action.yml:1]: Same pattern in https://github.com/actions/download-artifact, so I'll apply your suggestion, thanks for this :)

File: buildkite/download-artifact/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          token: ${{ secrets.BUILDKITE_TOKEN }}
+          pipeline: "my-super-pipeline"
+          wait-for: true
+
+      - uses: elastic/oblt-actions/buildkite/download-artifact@v1
+        with:
+          build: ${{ steps.buildkite-run.outputs.build }}
+          path: "artifacts.tar"
+          pipeline: "my-super-pipeline"
```
Comment (reakaleek) [inline buildkite/download-artifact/README.md:35]: should we add the output `pipeline` to the `buildkite/run` action?

so we can do 

```
pipeline: ${{ steps.build-run.outputs.pipeline }}
```

and only need to specify it once.

In case the pipeline name is changed in the future.. you only need to do it in a single place.

**Review by reakaleek — APPROVED** [review]

File: buildkite/download-artifact/README.md (hunk 3 above)
Comment (v1v) [reply buildkite/download-artifact/README.md:35]: I like that

File: buildkite/download/action.yml (hunk 2 above)
Comment (v1v) [reply buildkite/download/action.yml]: see [`5e8e6e6` (#18)](https://github.com/elastic/oblt-actions/pull/18/commits/5e8e6e6d8614da8adaf795511bf675eef47e542e)

File: buildkite/download-artifact/README.md (hunk 3 above)
Comment (v1v) [reply buildkite/download-artifact/README.md]: see [`697b0cf` (#18)](https://github.com/elastic/oblt-actions/pull/18/commits/697b0cf341fd4098ab5cab139d40db3cd6071aec)

**Review by reakaleek — APPROVED** [review]
